lib/wallet/close_channel.dart

```diff
@@ -0,0 +1,118 @@
+
+class CloseChannel extends StatefulWidget {
+  const CloseChannel({Key? key}) : super(key: key);
+
+  static const route = '/' + subRouteName;
+  static const subRouteName = 'close-channel';
+
+  @override
+  State<CloseChannel> createState() => _CloseChannelState();
+}
+
+class _CloseChannelState extends State<CloseChannel> {
+  bool submitting = false;
+  late AmountDisplay closeAmount;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    LightningBalance lightningBalance = context.watch<LightningBalance>();
+    closeAmount = lightningBalance.amount.display(currency: Currency.sat);
+
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Close channel'),
+      ),
+      body: SafeArea(
+          child: Stack(
+        children: [
+          Container(
+            padding: const EdgeInsets.all(20.0),
+            child: Column(children: [
+              Center(
+                child: RichText(
+                    text: TextSpan(
+                        style: const TextStyle(color: Colors.black, fontSize: 18),
+                        children: [
+                      const TextSpan(
+                          text: "This will close your Lightning channel with the 10101 maker\n\n",
+                          style: TextStyle(fontWeight: FontWeight.bold)),
+                      TextSpan(
+                          text:
+                              "Closing the channel will send ${closeAmount.value} sats to your 10101 on-chain wallet.\n\n"),
+                      const TextSpan(
+                          text:
+                              "Once the channel is closed, you will have to open a new one if you want to continue using your wallet for Lightning payments and trading.")
+                    ])),
+              ),
+            ]),
+          ),
+          Align(
+              alignment: Alignment.bottomCenter,
+              child: Container(
+                  padding: const EdgeInsets.all(20.0),
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.end,
+                    children: [
+                      AlertMessage(
+                          message: Message(
+                              title:
+                                  "Every time you open a new channel, you incur in additional transaction fees.",
+                              type: AlertType.info)),
+                      const SizedBox(height: 50),
+                    ],
+                  )))
+        ],
+      )),
+      floatingActionButton: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          Padding(
+              padding: const EdgeInsets.only(right: 20.0),
+              child: ElevatedButton.icon(
+                label: const Text('Confirm'),
+                icon: !submitting ? const SizedBox.shrink() : const SubmittingButtonIcon(),
```

Comment:
❓ is there a reason you are using SizedBox.shrink() instead of an empty Container? I think a container would have 0 width and 0 height by default?

Reply:
I tried using a `Container` at first, but it would cause the button to have the height of the whole screen.